### ISSUE TITLE
Feature: shared object between mutation and network

### ISSUE THREAD

**Issue by morrys** [facebook/relay#2804]
I wanted to propose the addition of an optional parameter in commitMutation shared with the network interface, in order to allow add extra information during the execution of the mutation.

This parameter will be very useful in offline contexts, but I think it is also useful for other use cases.

**Comment by sibelius** [comment]
Can’t you use updater?

**Comment by morrys** [comment]
No, because that function is not used by the network.

https://github.com/facebook/relay/blob/master/packages/relay-runtime/store/RelayModernEnvironment.js#L327

perhaps an example makes it clearer:

as is
```
const source = this._network.execute (
         operation.node.params,
         operation.variables,
         {force: true},
         uploadables,
       );
```

to be

```
const source = this._network.execute (
         operation.node.params,
         operation.variables,
         {force: true},
         uploadables,
         extra
       );
```

**Comment by josephsavona** [comment]
This is what what `CacheConfig`  (third arg) is intended for.  The `metadata` can be any object w whatever info  you need.

**Comment by morrys** [comment]
Great news, my problem is solved :) maybe it could be useful to add this property in commitMutation too:

```
commitMutation(environment, {
    mutation,
    variables: {
      input,
    },
    extra
})
```
and to modify RelayModernEnvironment to use it as you suggested:

https://github.com/facebook/relay/blob/master/packages/relay-runtime/store/RelayModernEnvironment.js#L330

`{force: true, metadata: extra }`

what do you think?

**Comment by josephsavona** [comment]
Yeah, totally reasonable to add CacheConfig as an optional arg to those methods. For executeMutation it’s a bit awkward though bc force:false doesn’t make sense.

**Comment by morrys** [comment]
Perhaps it is sufficient to create a subtype with a different name :)
